lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -141,14 +159,9 @@ def set_3d_properties(self, zs=0, zdir='z'):
         xs = self.get_xdata()
         ys = self.get_ydata()
 
-        try:
-            # If *zs* is a list or array, then this will fail and
-            # just proceed to juggle_axes().
-            zs = float(zs)
-            zs = [zs for x in xs]
-        except TypeError:
-            pass
-        self._verts3d = juggle_axes(xs, ys, zs, zdir)
+        zs = np.broadcast_to(zs, len(xs))
```

Comment:
Maybe just better to do `xs, ys, zs = np.broadcast_arrays(xs, ys, zs)`, which covers more cases

Reply:
Broadcasting only `zs` is simpler to follow IMO, unless you think there is a case when `xs`, `ys` also needs to be to be broadcasted? if there is a case when `len(xs) ! = len(ys)`, that is not covered by tests / examples? 

In general wouldn't it be preferable to explicitly fail with a shape mismatch if something goes wrong rather than let `np.broadcast_arrays` magically fix the shapes, but then the issue will be rather difficult to debug..